### PR TITLE
handle basic behaviour termination cleanly

### PR DESCRIPTION
**Pull request by gauthamm**
Handle termination of basic behaviour cleanly.

### CODE REVIEW

File: alica_engine/src/engine/BehaviourPool.cpp [hunk 1]
```diff
@@ -74,11 +74,26 @@ void BehaviourPool::stopAll()
...
     }
 }
 
+/**
+ * Calls terminate on all BasicBehaviours.
+ */
+void BehaviourPool::terminateAll()
+{
+    const PlanRepository::Accessor<BehaviourConfiguration>& behaviourConfs = _ae->getPlanRepository()->getBehaviourConfigurations();
+    for (const BehaviourConfiguration* beh : behaviourConfs) {
```
Comment (abhi1287) [inline alica_engine/src/engine/BehaviourPool.cpp]: You can directly iterate over the _availableBehaviours map. Example:

for( auto& behaviour : _availableBehaviours  ) {
        behaviour.second->terminate();
 }

Comment (abhi1287) [reply alica_engine/src/engine/BehaviourPool.cpp]: Also, the presence of both init / terminate and start / stop is confusing. I will push a separate patch to rename start / stop  to something else as essentially they are only signalling.
@hendrik-skubch  suggestions?

Comment (hendrik-skubch) [reply alica_engine/src/engine/BehaviourPool.cpp]: start and stop are aptly named in my opinion, as they start and stop the behavior.
Terminate is called at the end of a behaviors lifetime. Not sure, but maybe a good option is the following:
initialiseParameters => onStart()
start() => start()
stop() => stop()
onTerminate() => onStop()
terminate() => terminate()
initInternal => startInternal()

**Review by hendrik-skubch — APPROVED** [review]

**Review by abhi1287 — APPROVED** [review]
Noted

**Pull request merged**